src/source/RazorEngine.Core/Templating/TemplateBase.cs

```diff
@@ -237,23 +240,48 @@ void ITemplate.Run(ExecuteContext context, TextWriter reader)
 
                     if (layout == null)
                     {
-                        throw new ArgumentException("Template you are trying to run uses layout, but no layout found in cache or by resolver.");
+                        throw new ArgumentException(
+                            "Template you are trying to run uses layout, but no layout found in cache or by resolver.");
                     }
 
-                    // Push the current body instance onto the stack for later execution.
-                    var body = new TemplateWriter(tw => tw.Write(builder.ToString()));
-                    context.PushBody(body);
+                    var body = new TemplateWriter(
+                        tw =>
+                            {
+                                using (var reader = new StreamReader(tempFilePath))
+                                {
+                                    // Push the current body instance onto the stack for later execution.
+                                    while ( !reader.EndOfStream)
+                                    {
+                                        tw.Write(reader.ReadLine());
+                                    }
+                                }
+                            });
+                        
+                        context.PushBody(body);
+                    
                     context.PushSections();
 
 #if RAZOR4
-                    await layout.Run(context, reader);
+                    await layout.Run(context, outputWriter);
 #else
-                    layout.Run(context, reader);
+                    layout.Run(context, outputWriter);
 #endif
                     return;
                 }
 
-                reader.Write(builder.ToString());
+                using (var reader = new StreamReader(tempFilePath))
+                {
+                    // Push the current body instance onto the stack for later execution.
+                    while (!reader.EndOfStream)
+                    {
+                        outputWriter.Write(reader.ReadLine());
```

Comment:
Many thanks :)

I had a problem generating huge files, that's why I tried to use a file
stream instead of memory one.

Does that change look sensible?

Possibly the best approach would be to generate small parts of the text at
once and append them to file.

Cheers,
AD


On Sat, Jul 16, 2016 at 5:20 PM, campersau <notifications@github.com> wrote:

> In src/source/RazorEngine.Core/Templating/TemplateBase.cs
> <https://github.com/Antaris/RazorEngine/pull/393#discussion_r71067372>:
>
> >  #endif
> >                      return;
> >                  }
> >
> > -                reader.Write(builder.ToString());
> > +                using (var reader = new StreamReader(tempFilePath))
> > +                {
> > +                    // Push the current body instance onto the stack for later execution.
> > +                    while (!reader.EndOfStream)
> > +                    {
> > +                        outputWriter.Write(reader.ReadLine());
>
> ReadLine does not include the new line characters. That's why some tests
> are failing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Antaris/RazorEngine/pull/393/files/605e08f364f35515d3f4da32ab4a0164a493ce5b#r71067372>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAyu5uoPDIymOAH48P8qLWl_oZ3sW7tRks5qWPamgaJpZM4I5pRR>
> .
>